File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {

        int[] answer = new int[nums1.length];

        int idx1 = 0;
        int idx2 = 0;

        int idx = 0;
        int val = 0;
        while(idx1 < m && idx2 <n ) {
            if(nums1[idx1] < nums2[idx2]) {
                val = nums1[idx1++];
            } else {
                val = nums2[idx2++];
            }
            answer[idx++] = val;
        }

        while(idx1 < m) {
            answer[idx++] = nums1[idx1++];
        }
        while(idx2 < n) {
            answer[idx++] = nums2[idx2++];
        }

        for(int i = 0; i < nums1.length; i++) {
            nums1[i] = answer[i];
        }

    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
#include <iostream>
using namespace std;


class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int size = m + n;        
        int* arr = new int[size];

        int index1 = 0;
        int index2 = 0;
        int p = 0;
        while(index1 < m && index2 < n) {
            if(nums1[index1] < nums2[index2]) {
                arr[p++] = nums1[index1++];
            } else {
                arr[p++] = nums2[index2++];
            }
        }    

        while(index1 < m) {
            arr[p++] = nums1[index1++];
        }
        while(index2 < n) {
            arr[p++] = nums2[index2++];
        }

        for(int i = 0; i < size; i++)
            nums1[i] = arr[i];

        delete[] arr;
        arr = nullptr;
    }
};